Sources/ToucanSource/BuildTargetSourceLoader.swift

```diff
@@ -177,17 +176,30 @@ public struct BuildTargetSourceLoader {
 
     /// Loads the main configuration object from the source.
     ///
+    /// The loader first attempts to load a configuration file named `config-{target.name}`. 
+    /// If that fails, it falls back to loading the default `config` file from the same source location.
+    /// If both attempts fail, a `SourceLoaderError` is thrown with detailed context.
+    ///
     /// - Returns: A `Config` object loaded from the source.
     /// - Throws: A `SourceLoaderError` if loading fails.
     func loadConfig() throws(SourceLoaderError) -> Config {
         do {
             let configUrl = sourceUrl.appendingPathIfPresent(target.config)
-            let config = try load(
-                type: Config.self,
-                named: "config",
-                at: configUrl
-            )
-            return config
+
+            do {
+                return try load(
+                    type: Config.self,
+                    named: "config-\(target.name)",
+                    at: configUrl
+                )
+            }
+            catch {
+                return try load(
```

Comment:
Not sure if falling back to the default is a good practice here... This could lead to unwanted side-effects e.g. user wants to load config-dev, but it's invalid, so the system will load config instead, but that's not all right for that target.